src/controllers/hooks.js

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import wrap from '@adobe/helix-shared-wrap';
+import { Message, Blocks, Elements } from 'slack-block-builder';
+import { internalServerError, notFound, ok } from '@adobe/spacecat-shared-http-utils';
+import { composeBaseURL, hasText } from '@adobe/spacecat-shared-utils';
+
+import { BaseSlackClient, SLACK_TARGETS } from '@adobe/spacecat-shared-slack-client';
+import { SITE_CANDIDATE_STATUS, SITE_CANDIDATE_SOURCES } from '@adobe/spacecat-shared-data-access/src/models/site-candidate.js';
+import { fetch } from '../support/utils.js';
+
+const CDN_HOOK_SECRET_NAME = 'INCOMING_WEBHOOK_SECRET_CDN';
+const RUM_HOOK_SECRET_NAME = 'INCOMING_WEBHOOK_SECRET_RUM';
+
+export const BUTTON_LABELS = {
+  APPROVE_CUSTOMER: 'As Customer',
+  APPROVE_FRIENDS_FAMILY: 'As Friends/Family',
+  IGNORE: 'Ignore',
+};
+
+const IGNORED_SUBDOMAIN_TOKENS = ['demo', 'dev', 'stag', 'qa', '--'];
```

Comment:
stage or stg?

Reply:
used `stag` to capture both `stage` and `staging` > both seem to exist in the logs

Reply:
_stag_: a male deer, especially a male red deer after its fifth year.